aiida/engine/processes/workchains/restart.py

```diff
@@ -251,20 +293,25 @@ def __init__(self, *args, **kwargs):
         if self._process_class is None or not issubclass(self._process_class, Process):
             raise ValueError('no valid Process class defined for `_process_class` attribute')
 
-    def _handlers(self):
-        """Return the list of all methods decorated with the `process_handler` decorator.
+    @classmethod
+    def is_process_handler(cls, process_handler_name):
+        """Return whether the given method name corresponds to a process handler of this class.
 
-        :return: list of process handler methods
+        :param process_handler_name: string name of the instance method
+        :return: boolean, True if corresponds to process handler, False otherwise
         """
-        from .utils import process_handler
-
-        handlers = []
+        # pylint: disable=comparison-with-callable
+        if isinstance(process_handler_name, str):
+            handler = getattr(cls, process_handler_name, {})
+        else:
+            handler = process_handler_name
 
-        for method in inspect.getmembers(self, predicate=inspect.ismethod):
-            if hasattr(method[1], 'decorator') and method[1].decorator == process_handler:  # pylint: disable=comparison-with-callable
-                handlers.append(method[1])
+        return getattr(handler, 'decorator', None) == process_handler
 
-        return handlers
+    @classmethod
+    def get_process_handlers(cls):
```

Comment:
> Not sure if this is easy to do, but it would be nice to show the parameters specified

You can in fact, because `get_process_handler` returns the _decorated_ function where the parameters of the decorator have been set as attributes, i.e. you can do the following:
```
In [4]: ArithmeticAddBaseWorkChain.get_process_handlers()[0].priority                                                                                                        
Out[4]: 450
```
This also allows you to sort them
```
sorted(ArithmeticAddBaseWorkChain.get_process_handlers(), key=lambda x: x.priority)
```

Reply:
alright, thanks!